solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -451,6 +451,67 @@ contract LiquidityProtection is TokenHandler, Utils, Owned, ReentrancyGuard, Tim
         return addProtectedLiquidity(_owner, poolToken, _baseToken, poolTokenAmount / 2, _amount);
     }
 
+    /**
+     * @dev returns the single-side staking limits of a given pool
+     *
+     * @param _poolAnchor   anchor of the pool
+     * @return maximum amount of base tokens that can be single-side staked in the pool
+     * @return maximum amount of network tokens that can be single-side staked in the pool
+     */
+    function poolAvailableSpace(IConverterAnchor _poolAnchor) external view returns (uint256, uint256) {
+        return (baseTokenAvailableSpace(_poolAnchor), networkTokenTokenAvailableSpace(_poolAnchor));
+    }
+
+    /**
+     * @dev returns the base-token staking limits of a given pool
+     *
+     * @param _poolAnchor   anchor of the pool
+     * @return maximum amount of base tokens that can be single-side staked in the pool
+     */
+    function baseTokenAvailableSpace(IConverterAnchor _poolAnchor) internal view returns (uint256) {
```

Comment:
might as well make these functions public to increase flexility.
also, since they are separated, we can re-use them to save some code (unless it's not worth it gas-wise).

Reply:
I've considered that (using each one of the getter functions in the corresponding setter function), however, it would actually increase the gas-cost of each one of the setter functions, since some of the storage read operations and the external function calls would actually be duplicated (i.e., both in the getter functions and in the setter functions).

We can find a way to flatten all of that (e.g., using several private functions which will receive the common data and process it), but it will uglify the code significantly, and we will possibly find ourselves dealing with 'stack too deep' and other issues.

In short, it might require a lot of work to do code-reuse here, which won't reduce the contract bytecode size (if that's what you're aiming at), and can even pose a certain risk - we can live with a bug in the getter function, but not if it's used in by a setter function.

Reply:
`might as well make these functions public to increase flexibility` - done.